### PR TITLE
Refactor Duplicated Code

### PR DESCRIPTION
**Pull request by maennchen**
* [x] Merge #123 first

### CODE REVIEW

File: lib/paper_trail.ex [hunk 1]
```diff
@@ -85,65 +76,55 @@ defmodule PaperTrail do
...
+  end
+
+  @spec model_or_error({:ok, %{model: model}}) :: model when model: struct()
+  defp model_or_error({:ok, %{model: model}}) do
+    model
+  end
+
+  @spec model_or_error({:error, reason :: term}) :: no_return
+  defp model_or_error({:error, %Ecto.Changeset{} = changeset}) do
+    raise Ecto.InvalidChangesetError, action: :update, changeset: changeset
```
Comment (maennchen) [inline lib/paper_trail.ex:124]: https://github.com/izelnakri/paper_trail/pull/112#discussion_r542908097

Comment (maennchen) [reply lib/paper_trail.ex]: @izelnakri Solved in 1aaf46279233c53ef38f58dde456210ae045ab6b

**Comment by izelnakri** [comment]
@maennchen merged the first 2 PRs, this one also looks good to me. Could you fix the merge conflict and push the changes, then I'll merge it. Thanks!

**Comment by maennchen** [comment]
@izelnakri Rebased.

**Pull request merged**